### PR TITLE
26 remove unneeded options

### PR DESCRIPTION
**Pull request by heshamMassoud**
#### Summary

1. Remove removeOtherSetEntries, removeOtherCollectionEntries and removeOtherProperties options. 3e3ec28

2. Remove removeOtherVariants option. 09504dd

#### Relevant Issues
#26, #128, #51

#### Todo
@butenkor do you think it makes sense to add any any of the options in these issues (#26, #128, #51) as templates?

- Tests
    - [x] Unit removed
    - ~[ ] Integration~ not needed
- [x] Documentation removed
- [x] Add Release Notes entry.

### CODE REVIEW

File: docs/RELEASE_NOTES.md [hunk 1]
```diff
@@ -55,13 +55,14 @@
 **Bug Fixes** (1)
 - **Commons** - Fixed library version in User-Agent headers of JVM SDK clients using the library. [#191](https://github.com/commercetools/commercetools-sync-java/issues/191)
 
-**Migration guide** (4)
+**Migration guide** (6)
+- **Product Sync** - Removed `removeOtherVariants` option which is already done by the sync by default. [#26](https://github.com/commercetools/commercetools-sync-java/issues/26)
```
Comment (butenkor) [inline docs/RELEASE_NOTES.md]: `option which is already done by the sync by default. Removal of variants can be prevented through beforeCreate and beforeUpdate callbacks. For more examples see here...`

File: docs/RELEASE_NOTES.md [hunk 2]
```diff
@@ -55,13 +55,14 @@
...
+- **Commons** - Removed `removeOtherSetEntries`, `removeOtherCollectionEntries` and `removeOtherProperties`. [#26](https://github.com/commercetools/commercetools-sync-java/issues/26)
```
Comment (butenkor) [inline docs/RELEASE_NOTES.md]: Same a above

File: src/main/java/com/commercetools/sync/products/ProductSyncOptions.java [hunk 3]
```diff
@@ -17,38 +17,28 @@
 import static java.util.Optional.ofNullable;
 
 /**
- * TODO: removeOtherVariants, whiteList, blackList, auto publish, revert staged changes, update staged.
+ * TODO: auto publish, revert staged changes, update staged.
```
Comment (butenkor) [inline src/main/java/com/commercetools/sync/products/ProductSyncOptions.java]: we already use staged by default

File: docs/RELEASE_NOTES.md (hunk 1 above)
Comment (heshamMassoud) [reply docs/RELEASE_NOTES.md]: c2dca78 I added the [KeepOtherVariantsSync](https://github.com/commercetools/commercetools-sync-java/pull/196/files#diff-8b48435be042780bac4308837e491ea2) as a template and an [integration test](https://github.com/commercetools/commercetools-sync-java/pull/196/files#diff-c2377e77a323178f42915dc71af1a73d) for it that the release notes [links to](c2dca78).

File: docs/RELEASE_NOTES.md (hunk 2 above)
Comment (heshamMassoud) [reply docs/RELEASE_NOTES.md]: c2dca78 I added the [KeepOtherVariantsSync](https://github.com/commercetools/commercetools-sync-java/pull/196/files#diff-8b48435be042780bac4308837e491ea2) as a template and an [integration test](https://github.com/commercetools/commercetools-sync-java/pull/196/files#diff-c2377e77a323178f42915dc71af1a73d) for it that the release notes [links to](c2dca78).

File: src/main/java/com/commercetools/sync/products/ProductSyncOptions.java (hunk 3 above)
Comment (heshamMassoud) [reply src/main/java/com/commercetools/sync/products/ProductSyncOptions.java]: 703ab89

File: docs/RELEASE_NOTES.md [hunk 4]
```diff
@@ -55,13 +55,19 @@
 **Bug Fixes** (1)
 - **Commons** - Fixed library version in User-Agent headers of JVM SDK clients using the library. [#191](https://github.com/commercetools/commercetools-sync-java/issues/191)
 
-**Migration guide** (4)
+**Migration guide** (6)
+- **Product Sync** - Removed `removeOtherVariants` option which is already done by the sync by default. Removal of 
+variants can be prevented through the beforeUpdateCallback. Please check [here](/src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/products/templates/beforeupdatecallback/KeepOtherVariantsSyncIT.java)
+ for an example of how this can be done. [#26](https://github.com/commercetools/commercetools-sync-java/issues/26)
+- **Commons** - Removed `removeOtherSetEntries`, `removeOtherCollectionEntries` and `removeOtherProperties` options 
+which are already done by the sync by default. Such options can be disbaled through the beforeUpdateCallback. Please
```
Comment (butenkor) [inline docs/RELEASE_NOTES.md]: `Such options can be disbaled through the beforeUpdateCallback` -> `Mentioned options and even more use cases can now be covered with help of beforeCreateCallback and beforeUpdateCallback`

File: src/main/java/com/commercetools/sync/products/templates/beforeupdatecallback/KeepOtherVariantsSync.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+     * @param oldProduct      the old existing {@link Product}.
+     * @return the same list of supplied {@code updateActions} without {@link RemoveVariant} update actions.
+     */
+    public static List<UpdateAction<Product>> keepOtherVariants(
+        @Nonnull final List<UpdateAction<Product>> updateActions,
+        @Nonnull final ProductDraft newProductDraft,
+        @Nonnull final Product oldProduct) {
+        return updateActions.stream()
+                            .filter(updateAction -> !(updateAction instanceof RemoveVariant))
+                            .collect(Collectors.toList());
```
Comment (butenkor) [inline src/main/java/com/commercetools/sync/products/templates/beforeupdatecallback/KeepOtherVariantsSync.java]: 👍

File: docs/RELEASE_NOTES.md (hunk 4 above)
Comment (heshamMassoud) [reply docs/RELEASE_NOTES.md]: f4b29d9

**Pull request merged**